src/librustc_codegen_llvm/mir/block.rs

```diff
@@ -462,6 +462,55 @@ impl FunctionCx<'a, 'll, 'tcx> {
                     return;
                 }
 
+                if (intrinsic == Some("init") || intrinsic == Some("uninit")) &&
+                    bx.cx.layout_of(sig.output()).abi.is_uninhabited()
```

Comment:
Can you move this below `let fn_ty`? Then you can use `fn_ty.ret.layout` instead of calling `layout_of`.